rlclientlib/serialization/payload_serializer.h

```diff
@@ -0,0 +1,134 @@
+        flatbuffers::FlatBufferBuilder fbb;
+        std::vector<flatbuffers::Offset<v2::SlotEvent>> slots;
+        for (size_t i = 0; i < action_ids.size(); i++)
+        {
+          slots.push_back(v2::CreateSlotEventDirect(fbb, &action_ids[i], &pdfs[i]));
+        }
+
+        std::vector<unsigned char> _context;
+        std::string context_str(context);
+        copy(context_str.begin(), context_str.end(), std::back_inserter(_context));
+
+        auto fb = v2::CreateCcbEventDirect(fbb, &_context, &slots, model_version.c_str(), flags | action_flags::DEFERRED);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+    };
+
+    struct slates_serializer : payload_serializer<payload_type::SLATES> {
+      static generic_event::payload_buffer_t event(const char* context, unsigned int flags, const std::vector<std::vector<uint32_t>>& action_ids,
+        const std::vector<std::vector<float>>& pdfs, const std::string& model_version) {
+        flatbuffers::FlatBufferBuilder fbb;
+        std::vector<flatbuffers::Offset<v2::SlatesSlotEvent>> slots;
+        for (size_t i = 0; i < action_ids.size(); i++)
+        {
+          slots.push_back(v2::CreateSlatesSlotEventDirect(fbb, &action_ids[i], &pdfs[i]));
+        }
+
+        std::vector<unsigned char> _context;
+        std::string context_str(context);
+        copy(context_str.begin(), context_str.end(), std::back_inserter(_context));
+
+        auto fb = v2::CreateSlatesEventDirect(fbb, &_context, &slots, model_version.c_str(), flags | action_flags::DEFERRED);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+    };
+
+    struct outcome_single_serializer : payload_serializer<payload_type::OUTCOME_SINGLE> {
+      static generic_event::payload_buffer_t event(float outcome) {
+        flatbuffers::FlatBufferBuilder fbb;
+        const auto evt = v2::CreateNumericEventSingle(fbb, outcome).Union();
+        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_NumericEventSingle, evt);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+
+      static generic_event::payload_buffer_t event(const char* outcome) {
+        flatbuffers::FlatBufferBuilder fbb;
+        const auto evt = v2::CreateStringEventSingleDirect(fbb, outcome).Union();
+        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_StringEventSingle, evt);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+
+      static generic_event::payload_buffer_t event(int index, float outcome) {
+        flatbuffers::FlatBufferBuilder fbb;
+        const auto evt = v2::CreateNumericEventIndexed(fbb, outcome, index).Union();
+        auto fb = v2::CreateOutcomeSingleEvent(fbb, v2::OutcomeSingleEventBody_NumericEventIndexed, evt);
+        fbb.Finish(fb);
+        return fbb.Release();
+      }
+
+      static generic_event::payload_buffer_t event(int index, const char* outcome) {
```

Comment:
Does C++ special case 0 as convertible to NULL and thus maybe leading to some weird overloading resolution conflict here?